gateway/ln-gateway/src/auth_manager.rs

```diff
@@ -0,0 +1,59 @@
+use std::time::UNIX_EPOCH;
+
+use fedimint_core::secp256k1::PublicKey;
+use jsonwebtoken::{encode, EncodingKey, Header};
+use serde::{Deserialize, Serialize};
+
+const SESSION_EXPIRY_SECONDS: u64 = 60 * 30; // 30 minute
+
+pub struct AuthManager {
+    ///gateway id
+    gateway_id: PublicKey,
+    /// A secret key to encode a JWT with
+    pub encoding_secret: [u8; 16],
```

Comment:
Why is this pub? Also could you make that an `EncodingKey`, so you don't have to run `EncodingKey::from_secret` each time a session is generated?